code/modules/clothing/glasses/glasses.dm

```diff
@@ -232,57 +233,47 @@
 	flags_inventory = COVEREYES
 	flags_inv_hide = HIDEEYES
 	eye_protection = 2
+	activation_sound = null
+	deactivation_sound = null
 
 /obj/item/clothing/glasses/welding/Initialize(mapload)
 	. = ..()
 	AddComponent(/datum/component/clothing_tint, TINT_5, TRUE)
 
-/obj/item/clothing/glasses/welding/proc/flip_up()
-	DISABLE_BITFIELD(flags_inventory, COVEREYES)
-	DISABLE_BITFIELD(flags_inv_hide, HIDEEYES)
-	DISABLE_BITFIELD(flags_armor_protection, EYES)
-	eye_protection = 0
-	icon_state = "[initial(icon_state)]up"
-
-/obj/item/clothing/glasses/welding/proc/flip_down()
-	ENABLE_BITFIELD(flags_inventory, COVEREYES)
-	ENABLE_BITFIELD(flags_inv_hide, HIDEEYES)
-	ENABLE_BITFIELD(flags_armor_protection, EYES)
-	eye_protection = initial(eye_protection)
-	icon_state = initial(icon_state)
-
 /obj/item/clothing/glasses/welding/verb/verbtoggle()
 	set category = "Object"
 	set name = "Adjust welding goggles"
 	set src in usr
 
 	if(!usr.incapacitated())
-		toggle_item_state(usr)
-
-/obj/item/clothing/glasses/welding/attack_self(mob/user)
-	toggle_item_state(user)
+		activate(usr)
 
-/obj/item/clothing/glasses/welding/toggle_item_state(mob/user)
-	. = ..()
-	active = !active
-	icon_state = "[initial(icon_state)][!active ? "up" : ""]"
-	if(!active)
-		flip_up()
+/obj/item/clothing/glasses/welding/activate(mob/user)
+	if(active)
+		DISABLE_BITFIELD(flags_inventory, COVEREYES)
```

Comment:
you can toggle bitfields instead of doing if, else, which is consistant with what you're doing elsewhere, but honestly the whole thing is better to have a turn on and turf on proc instead of toggling, I don't know why you changed it, as its less safe and requires extra checks if you want to set to a specific state instead of just calling that proc directly.

Reply:
I thought it was much better to have one `activate` proc when all you're really doing with glasses is toggling functions. Do you want me to make them all use `turn_on` and `turn_off` procs?

Reply:
yes, see below.